services/scabbard/src/cli/transaction.rs

```diff
@@ -0,0 +1,402 @@
+            let org_id = delete_smart_permission.org_id();
+            let name = delete_smart_permission.name();
+            let addresses = vec![
+                compute_smart_permission_address(org_id, name)?,
+                compute_org_address(org_id)?,
+                compute_agent_address(&public_key)?,
+            ];
+
+            (addresses.clone(), addresses)
+        }
+    };
+
+    txn_header.set_inputs(protobuf::RepeatedField::from_vec(input_addresses));
+    txn_header.set_outputs(protobuf::RepeatedField::from_vec(output_addresses));
+
+    let payload_bytes = payload.into_bytes()?;
+    let hash = HashSigner.sign(&payload_bytes).map_err(|err| {
+        CliError::action_error_with_source("failed to hash payload bytes", err.into())
+    })?;
+    txn_header.set_payload_sha512(to_hex(&hash));
+    txn.set_payload(payload_bytes);
+
+    let txn_header_bytes = txn_header.write_to_bytes()?;
+    txn.set_header(txn_header_bytes.clone());
+
+    let b: &[u8] = &txn_header_bytes;
+    let signature = signer.sign(b)?;
+    txn.set_header_signature(to_hex(&signature));
+
+    Ok(txn)
+}
+
+/// Returns a Batch for the given Transactions and Signer
+///
+/// # Arguments
+///
+/// * `txns` - transaction to put in the batch
+/// * `signer` - the signer to be used to sign the batch
+///
+/// # Errors
+///
+/// An error is returned if it occurs during serialization of the provided transactions or
+/// internally created `BatchHeader`, or if a signing error occurs.
+pub fn create_batch(txns: Vec<Transaction>, signer: &dyn Signer) -> Result<Batch, CliError> {
+    let public_key = to_hex(signer.public_key());
+
+    let mut batch = Batch::new();
+    let mut batch_header = BatchHeader::new();
+
+    batch_header.set_transaction_ids(protobuf::RepeatedField::from_iter(
+        txns.iter().map(|txn| txn.header_signature.clone()),
+    ));
+    batch_header.set_signer_public_key(public_key);
+    batch.set_transactions(txns.into());
+
+    let batch_header_bytes = batch_header.write_to_bytes()?;
+    batch.set_header(batch_header_bytes.clone());
+
+    let b: &[u8] = &batch_header_bytes;
+    let signature = signer.sign(b)?;
+    batch.set_header_signature(to_hex(&signature));
+
+    Ok(batch)
+}
+
+/// Returns a BatchList containing the provided Batch
+///
+/// # Arguments
+///
+/// * `batch` - a Batch
+pub fn create_batch_list_from_one(batch: Batch) -> BatchList {
+    let mut batch_list = BatchList::new();
+    batch_list.set_batches(protobuf::RepeatedField::from_vec(vec![batch]));
+    batch_list
+}
+
+/// Creates a nonce appropriate for a TransactionHeader
+fn create_nonce() -> String {
+    let elapsed = Instant::now().elapsed();
+    format!("{}{}", elapsed.as_secs(), elapsed.subsec_nanos())
+}
+
+/// Returns a state address for a given namespace registry
+///
+/// # Arguments
+///
+/// * `namespace` - the address prefix for this namespace
+fn compute_namespace_registry_address(namespace: &str) -> Result<String, CliError> {
```

Comment:
I feel like this should be part of the builder sequence (same for similar fns).

Reply:
Created a story to implement this capability